src/taskpane/server.js

```diff
@@ -1949,19 +2255,87 @@ function manageSendingToSpira(entry, user, projectId, artifact, fields, fieldTyp
+              output.errorMessage = error;
+            } else {
+              output.errorMessage = "send attempt failed";
+            }
+
+            // reset the parentId if we are not on a subType - to make sure subTypes are not added to the wrong parent
+            if (artifact.hasSubType && !entry.isSubType) {
+              output.parentId = 0;
+            }
+            return output;
+          });
+
+      }
+      else {
+        return putArtifactToSpira(entry, user, projectId, artifactTypeIdToSend, parentId)
+          .then(function (response) {
+            var errorStatus = response.error;
+            if (!errorStatus) {
+              // get the id/subType id of the updated artifact
+              var artifactIdField = getIdFieldName(fields, fieldTypeEnums, entry.isSubType);
+              //just repeat the id - it's the same 
+              output.newId = entry[artifactIdField];
+              // repeats the output parent ID only if the artifact has a subtype and this entry is NOT a subtype
+              if (artifact.hasSubType && !entry.isSubType) {
+                output.parentId = parentId;
+              }
+              return output;
+
+            }
+          })
+          .catch(function (error) {
+            //we have an error - so set the flag and the message
+            output.error = true;
+            if (error) {
+              output.errorMessage = error;
+            } else {
+              output.errorMessage = "update attempt failed";
+            }
+
+            // reset the parentId if we are not on a subType - to make sure subTypes are not added to the wrong parent
+            if (artifact.hasSubType && !entry.isSubType) {
+              output.parentId = 0;
+            }
+            return output;
+          });
+      }
+    }
+    else if (isComment) {
+      //comment creation
+      return postArtifactToSpira(entry, user, projectId, artifactTypeIdToSend, parentId, isComment, false)
         .then(function (response) {
           output.fromSpira = response.body;
 
-          // get the id/subType id of the newly created artifact
-          var artifactIdField = getIdFieldName(fields, fieldTypeEnums, entry.isSubType);
-          output.newId = output.fromSpira[artifactIdField];
+          // // get the id/subType id of the newly created artifact
+          // var artifactIdField = getIdFieldName(fields, fieldTypeEnums, entry.isSubType);
+          // output.newId = output.fromSpira[artifactIdField];
 
-          // update the output parent ID to the new id only if the artifact has a subtype and this entry is NOT a subtype
-          if (artifact.hasSubType && !entry.isSubType) {
-            output.parentId = output.newId;
-          }
+          // // update the output parent ID to the new id only if the artifact has a subtype and this entry is NOT a subtype
+          // if (artifact.hasSubType && !entry.isSubType) {
+          //   output.parentId = output.newId;
+          // }

```

Comment:
Is this code still required or can it be deleted?

Reply:
It can be deleted - all these functionalities are handled by functions now.